lib/scholar/cluster/gmm.ex

```diff
@@ -0,0 +1,407 @@
+defmodule Scholar.Cluster.GaussianMixture do
+  @moduledoc """
+  Gaussian Mixture Model.
+
+  Gaussian Mixture model is a probabilistic model that assumes every data point is generated
+  by randomly choosing one of the several fixed Gaussian distributions and then sampling from it.
+  Its parameters are estimated using the Expectation-Maximization (EM) algorithm, which is an
+  iterative algorithm alternating between the two steps: the E-step which computes the
+  expectation of the Gaussian assignment for each data point x and the M-step which updates the
+  parameters to maximize the expectations found in E-step. While every iteration of the algorithm
+  is guaranteed to improve the log-likelihood, the final result depends on the initial values of
+  the parameters and the entire procedure should be repeated several times.
```

Comment:
From reading without looking at the code beforehand:

Does this mean that we need to run the algorithm multiple times, or does the algorithm itself repeat this procedure?

Reply:
The algorithm itself repeats the procedure `num_runs` amount of times, each time with different initial parameters. Similarly as it was already done in k-means implementation.

Reply:
I am not sure if I should change anything here. Thoughts?

Reply:
For me, the description is clear. @polvalente do you have any suggestions?